packages/go_router/lib/src/delegate.dart

```diff
@@ -166,6 +166,18 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     notifyListeners();
   }
 
+  /// Calls [pop] repeatedly until the predicate returns true.
+  void popUntil(bool Function(RouteMatch) predicate) {
+    bool hasPopped = false;
+    while (!predicate(_matchList.last)) {
```

Comment:
things has changed a bit, its probably the best to call pop directly since it will remove any pageless route that are on top such as dialog.

Reply:
@chunhtai It's hard to use go router.... Us project previous use more ‘pop with value on pages’, now have the requirement need do the web compatible, but go router no this feature to provide, then use 'Go_Router_Flow' lib, but looks like Pop function have been changed....

Reply:
No I mean the implementation of popUntil should use `pop` method. I am on board with implementing `popUntil`

Reply:
@chunhtai I am calling pop in [♻️ Call pop repeatedly](https://github.com/flutter/packages/pull/2728/commits/f9e08c21d8ef986cef270330385df7f270b40322)